scispacy/genia_tokenizer.py

```diff
@@ -0,0 +1,355 @@
+        sentence = sentence.replace(ptb_symbol, original)
+    return sentence
+
+# processing in three stages: "initial" regexs run first, then
+# "repeated" run as long as there are changes, and then "final"
+# run. As the tokenize() function itself is trivial, comments relating
+# to regexes given with the re.compiles.
+
+_INITIAL = []
+_REPEATED = []
+_FINAL = []
+
+# separate but do not break ellipsis
+_INITIAL.append((re.compile(r'\.\.\.'), r' ... '))
+#__initial.append((re.compile(r'(\n+)'), r' \1 '))
+
+# To avoid breaking names of chemicals, protein complexes and similar,
+# only add space to related special chars if there's already space on
+# at least one side.
+_INITIAL.append((re.compile(r'([,;:@#]) '), r' \1 '))
+_INITIAL.append((re.compile(r' ([,;:@#])'), r' \1 '))
+
+# always separated
+_INITIAL.append((re.compile(r'\$'), r' $ '))
+_INITIAL.append((re.compile(r'\%'), r' % '))
+_INITIAL.append((re.compile(r'\&'), r' & '))
+
+# separate punctuation followed by space even if there's closing
+# brackets or quotes in between, but only sentence-final for
+# periods (don't break e.g. "E. coli").
+# \u2019 is a alternate apostrophe that crops up in some papers.
+_INITIAL.append((re.compile(r'([\u2019,:;])([\[\]\)\}\>\"\']* +)', re.UNICODE), r' \1\2'))
+_INITIAL.append((re.compile(r'(\.+)([\[\]\)\}\>\"\']* +)$'), r' \1\2'))
+
+# these always
+_INITIAL.append((re.compile(r'\?'), ' ? '))
+_INITIAL.append((re.compile(r'\!'), ' ! '))
+
+# separate greater than and less than signs, avoiding breaking
+# "arrows" (e.g. "-->", ">>") and compound operators (e.g. "</=")
+_INITIAL.append((re.compile(r'((?:=\/)?<+(?:\/=|--+>?)?)'), r' \1 '))
+_INITIAL.append((re.compile(r'((?:<?--+|=\/)?>+(?:\/=)?)'), r' \1 '))
+
+# separate dashes, not breaking up "arrows"
+_INITIAL.append((re.compile(r'(<?--+\>?)'), r' \1 '))
+
+# Parens only separated when there's space around a balanced
+# bracketing. This aims to avoid splitting e.g. beta-(1,3)-glucan,
+# CD34(+), CD8(-)CD3(-).
+
+# Previously had a proper recursive implementation for this, but it
+# was much too slow for large-scale use. The following is
+# comparatively fast but a bit of a hack:
+
+# First "protect" token-internal brackets by replacing them with
+# their PTB escapes. "Token-internal" brackets are defined as
+# matching brackets of which at least one has no space on either
+# side. To match GTB tokenization for cases like "interleukin
+# (IL)-mediated", and "p65(RelA)/p50", treat following dashes and
+# slashes as space.  Nested brackets are resolved inside-out;
+# to get this right, add a heuristic considering boundary
+# brackets as "space".
+
+# (First a special case (rareish): "protect" cases with dashes after
+# paranthesized expressions that cannot be abbreviations to avoid
+# breaking up e.g. "(+)-pentazocine". Here, "cannot be abbreviations"
+# is taken as "contains no uppercase charater".)
+_INITIAL.append((re.compile(r'\(([^ ()\[\]{}]+)\)-'), r'-LRB-\1-RRB--'))
```

Comment:
I think the comment doesn't match the code here anymore. There is nothing in this regex that excludes uppercase characters.